src/main/scala/beam/sim/population/PopulationScaling.scala

```diff
@@ -195,6 +199,76 @@ trait PopulationScaling extends LazyLogging {
 
   }
 
+  def removeAgent(scenario: MutableScenario, beamConfig: BeamConfig): Unit = {
+    val removeAgent = beamConfig.beam.agentsim.agents.population.industryRemovalProbabilty.removeAgent
+    if (removeAgent) {
+      val industryFile = beamConfig.beam.agentsim.agents.population.industryRemovalProbabilty.inputFilePath
+      val rng: MersenneTwister = new MersenneTwister(beamConfig.matsim.modules.global.randomSeed)
+      if (new File(industryFile).exists()) {
+        removePerson(scenario, industryFile, rng)
+      } else {
+        logger.warn("Industry probability file not available!!!")
+      }
+    } else {
+      removeWorkPlan(scenario)
+    }
+  }
+
+  def removePerson(scenario: MutableScenario, filePath: String, rng: MersenneTwister): Unit = {
+    val (iter: Iterator[mutable.Map[String, String]], toClose: Closeable) =
+      GenericCsvReader.readAs[mutable.Map[String, String]](filePath, rec => rec.asScala, _ => true)
+    try {
+      val industrialProbability = iter.map(value => value("industry") -> value("removal_probability").toDouble).toMap
+      val persons = scenario.getPopulation.getPersons.values().asScala
+
+      //Build persons probability
+      val personDistributionPair = persons
+        .map(person => {
+          val industry = person.getAttributes.getAttribute("industry").toString
+          val probability = industrialProbability.getOrElse(industry, 0.0)
+          new PPair[Person, java.lang.Double](person, probability)
+        })
+        .toVector
+
+      val enumeratedDistribution = new EnumeratedDistribution[Person](rng, personDistributionPair.asJava)
+
+      //Calculate number of persons to be removed from population
+      val removalItemSize = persons
+        .groupBy(_.getAttributes.getAttribute("industry").toString)
```

Comment:
If there is no attribute `industry`, there will be NullPointerException. Let's handle that case

Reply:
handle NullPointerException